arches/app/src/components/ControlledListManager/ControlledListEditor.vue

```diff
@@ -0,0 +1,220 @@
+        arches.urls.controlled_list(props.displayedList.id)
+    );
+    await response.json().then((data) => {
+        items.value = data.items;
+    });
+};
+
+const createItem = async () => {
+    try {
+        const response = await fetch(arches.urls.controlled_list_item_add, {
+            method: "POST",
+            headers: {
+                "X-CSRFToken": Cookies.get("csrftoken"),
+            },
+            body: JSON.stringify({ list_id: props.displayedList.id }),
+        });
+        if (response.ok) {
+            const newItem = await response.json();
+            items.value.push(newItem);
+        } else {
+            throw new Error();
+        }
+    } catch {
+        toast.add({
+            severity: "error",
+            summary: $gettext("Item creation failed."),
+            life: 3000,
+        });
+    }
+};
+
+const deleteItems = async (selectedItems: ControlledList[]) => {
+    if (!selectedItems.length) {
+        return;
+    }
+    const promises = selectedItems.map((item) =>
+        fetch(arches.urls.controlled_list_item(item.id), {
+            method: "DELETE",
+            headers: {
+                "X-CSRFToken": Cookies.get("csrftoken"),
+            },
+        })
+    );
+
+    try {
+        const responses = await Promise.all(promises);
+        if (responses.some((resp) => !resp.ok)) {
+            throw new Error();
+        }
+    } catch {
+        toast.add({
+            severity: "error",
+            summary: $gettext("One or more items failed to delete."),
+            life: 3000,
+        });
+    }
+    await fetchItems();
+};
+</script>
+
+<template>
+    <ListHeader
+        :displayed-list="props.displayedList"
+        :is-item-editor="true"
+    />
+
+    <div class="list-editor-container">
+        <ListCharacteristics
+            :displayed-list="props.displayedList"
+            :editable="true"
+        />
+
+        <div
+            class="items"
+            style="margin: 1rem"
+        >
+            <h3 style="margin-top: 4rem; margin-left: 0">
+                {{ $gettext("List Item Editor") }}
+            </h3>
+            <Splitter
+                :pt="{
+                    gutter: { style: { background: lightGray } },
+                    gutterHandler: { style: { background: lightGray } },
+                }"
+            >
+                <SplitterPanel
+                    :size="30"
+                    :min-size="15"
+                >
+                    <Suspense>
+                        <SidepanelDataView
+                            :add-label="$gettext('Add New Item')"
+                            :delete-label="$gettext('Delete Item')"
+                            :delete-label-plural="$gettext('Delete Items')"
+                            :displayed-item="displayedItem"
+                            :create-item="createItem"
+                            :delete-items="deleteItems"
+                            :fetch-items="fetchItems"
```

Comment:
why are these methods defined here if they're only passed into a single child. Shouldn't they live in the child?

Reply:
fetch-items is also passed by <ControlledListInventory> to <SidepanelDataView>. The naming is obscuring that a little bit, because one sidepanel manages "lists" and one manages "items", so "fetch-items" is either getting "fetchLists" or "fetchItems". This is related to the comment about whether or not I should be angling to reuse components at all, will comment next there.

Reply:
`<SidepanelDataView>` is considerably less abstract in latest